packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -403,40 +404,90 @@ export class AccountTreeController extends BaseController<
   }
 
   /**
-   * Gets the first account ID in the specified group.
+   * Gets account group.
+   *
+   * @param groupId - The account group ID.
+   * @returns The account group or undefined if not found.
+   */
+  #getAccountGroup(groupId: AccountGroupId): AccountGroupObject | undefined {
+    const found = Object.values(this.state.accountTree.wallets).find(
+      (wallet) => wallet.groups[groupId] !== undefined,
+    );
+
+    return found?.groups[groupId];
+  }
+
+  /**
+   * Gets the default account for specified group.
    *
    * @param groupId - The account group ID.
    * @returns The first account ID in the group, or undefined if no accounts found.
    */
-  #getFirstAccountInGroup(groupId: AccountGroupId): AccountId | undefined {
-    for (const wallet of Object.values(this.state.accountTree.wallets)) {
-      if (wallet.groups[groupId]) {
-        const group = wallet.groups[groupId];
-        if (group && group.accounts.length > 0) {
-          return group.accounts[0];
+  #getDefaultAccountFromAccountGroupId(
+    groupId: AccountGroupId,
+  ): AccountId | undefined {
+    const group = this.#getAccountGroup(groupId);
+
+    if (group) {
+      let candidate;
+      for (const id of group.accounts) {
+        const account = this.messagingSystem.call(
+          'AccountsController:getAccount',
+          id,
+        );
+
+        if (!candidate) {
```

Comment:
```suggestion
        if (account && !candidate) {
```
Although I guess `AccountsController:getAccount` will always return an account, anyway.